File: backend/SNS.Application/Jobs/JobApplications/Contracts/JobApplicationDetailsDto.cs
using SNS.Application.Jobs.Jobs.Contracts;
using SNS.Application.Profiles.Profiles.Contracts;
using SNS.Domain.QA.Enums;

namespace SNS.Application.Jobs.JobApplications.Contracts;

public sealed record JobApplicationDetailsDto(
    Guid Id,
    Guid JobId,
    JobSnapshotDto Job,
    Guid ApplicantId,
    ProfileSnapshotDto Applicant,
    Guid? ResumeId,
    string CoverLetterText,
    string? ResumeFileUrl,
    ApplicationStatus Status,
    DateTime CreatedAt,
    DateTime UpdatedAt,
    bool IsActive
);

File: SNS.Web/src/app/jobs/job-applications/contracts/job-application-details.dto.ts
import { ProfileSnapshotDto } from '../../../profiles/profiles/contracts/profile-snapshot.dto';
import { ApplicationStatus } from '../../enums/application-status.enum';
import { JobSnapshotDto } from '../../jobs/contracts/job-snapshot.dto';

export interface JobApplicationDetailsDto {
    id: string;
    jobId: string;
    job: JobSnapshotDto;
    applicantId: string;
    applicant: ProfileSnapshotDto;
    resumeId: string | null;
    coverLetterText: string;
    resumeFileUrl: string | null;
    status: ApplicationStatus;
    createdAt: string;
    updatedAt: string;
    isActive: boolean;
}
